src/Tools/AI Test Toolkit/src/Logs/AITLogEntry.Table.al

```diff
@@ -0,0 +1,283 @@
+        if "Duration (ms)" = 0 then
+            "Duration (ms)" := "End Time" - "Start Time";
+    end;
+
+    procedure SetInputBlob(NewInput: Text)
+    var
+        OutStream: OutStream;
+    begin
+        Clear("Input Data");
+        "Input Data".CreateOutStream(OutStream, GetDefaultTextEncoding());
+        OutStream.Write(NewInput);
+    end;
+
+    procedure GetInputBlob(): Text
+    var
+        InStream: InStream;
+        InputContent: Text;
+    begin
+        CalcFields("Input Data");
+        "Input Data".CreateInStream(InStream, GetDefaultTextEncoding());
+        InStream.Read(InputContent);
+        exit(InputContent);
+    end;
+
+    procedure SetOutputBlob(NewOutput: Text)
+    var
+        OutStream: OutStream;
+    begin
+        Clear("Output Data");
+        "Output Data".CreateOutStream(OutStream, GetDefaultTextEncoding());
+        OutStream.Write(NewOutput);
+    end;
+
+    procedure GetOutputBlob(): Text
+    var
+        InStream: InStream;
+        OutputContent: Text;
+    begin
+        CalcFields("Output Data");
+        "Output Data".CreateInStream(InStream, GetDefaultTextEncoding());
+        InStream.Read(OutputContent);
+        exit(OutputContent);
+    end;
+
+    procedure SetMessage(NewMessageText: Text)
+    var
+        MessageOutStream: OutStream;
+    begin
+        Clear("Message Text");
+        "Message Text".CreateOutStream(MessageOutStream, GetDefaultTextEncoding());
+        MessageOutStream.WriteText(NewMessageText);
+    end;
+
+    procedure GetMessage(): Text
+    var
+        MessageInStream: InStream;
+        MessageText: Text;
+    begin
+        CalcFields("Message Text");
+        "Message Text".CreateInStream(MessageInStream, GetDefaultTextEncoding());
+        MessageInStream.ReadText(MessageText);
+        exit(MessageText);
+    end;
+
+    procedure SetErrorCallStack(ErrorCallStack: Text)
+    var
+        ErrorCallStackOutStream: OutStream;
+    begin
+        Clear("Error Call Stack");
+        "Error Call Stack".CreateOutStream(ErrorCallStackOutStream, GetDefaultTextEncoding());
+        ErrorCallStackOutStream.WriteText(ErrorCallStack);
+    end;
+
+    procedure GetErrorCallStack(): Text
+    var
+        ErrorCallStackInStream: InStream;
+        ErrorCallStackText: Text;
+    begin
+        CalcFields("Error Call Stack");
+        "Error Call Stack".CreateInStream(ErrorCallStackInStream, GetDefaultTextEncoding());
+        ErrorCallStackInStream.ReadText(ErrorCallStackText);
+        exit(ErrorCallStackText);
+    end;
+
+    local procedure GetDefaultTextEncoding(): TextEncoding
+    begin
+        exit(TextEncoding::UTF8);
+    end;
+
+    internal procedure SetFilterForFailedTestProcedures()
+    var
+        AITALTestSuiteMgt: Codeunit "AIT AL Test Suite Mgt";
+    begin
+        Rec.SetRange(Operation, AITALTestSuiteMgt.GetDefaultRunProcedureOperationLbl());
+        Rec.SetFilter("Procedure Name", '<> %1', '');
```

Comment:
No space required here: '<> %1' to '<>%1'

Reply:
Done